pkg/middleware/log_entry.go

```diff
@@ -24,3 +24,21 @@ func LogEntryMiddleware(middleware middlewareFunc, lggr *log.Logger, validatorHo
 		}
 	}
 }
+
+// LogContextMiddleware builds a log.Entry, setting the request fields
+// and storing it in the context to be used throughout the stack
+func LogContextMiddleware(lggr *log.Logger) buffalo.MiddlewareFunc {
```

Comment:
Can you remove the `LogEntryMiddleware` and rename this one to be `LogEntryMiddleware` and make sure the Filter is using this one properly. I'll update the Download Protocol logEntry in another PR :)

Reply:
@marwan-at-work For sure. I didn't actually apply the middleware or remove any working code because I wanted to validate the approach first. I actually made the changes in `pkg/download`, but didn't commit them for that reason. If you're good with this strategy though, I'm happy to commit those changes (here or in a separate PR), but if you'd rather take care of it, that's great too 👍